app/controllers/meteors.controller.ts

```diff
@@ -5,12 +5,17 @@ import queryValidation from "../middleware/queryValidation.js";
 import { meteorsQuerySchema } from "../schemas/meteors.schema.js";
 const router = express.Router();
 
+interface MeteorsQuery {
+  count?: string;
+  wereDangerousMeteors?: string;
+}
+
 router.get(
   "/meteors",
   queryValidation(meteorsQuerySchema, "query"),
   async (req, res, next) => {
-    const dates = validateDate(req?.query?.date);
-    const { count, wereDangerousMeteors } = req?.query || null;
+    const dates = validateDate(req?.query?.date as string | [string, string]);
```

Comment:
I'm not sure you need to cast it here
`validateDate` expects some unknown type and validate it inside